### ISSUE TITLE
Advantages of Anchor Free?

### ISSUE THREAD

**Issue by junghyun-avikus** [ultralytics/ultralytics#641]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Since I liked yolov5 a lot, I am quite excited to see yolov8 and can't wait to try it on my custom dataset.

One quick question is what are some advantages of anchor free models?
Since yolov2 have been using anchors, what made you move away from anchors?

**Comment by biendltb** [comment]
Few reasons for it:
* End-to-end differentiable: The network is trained end-to-end without involving any data processing (e.g. NMS)
* Better speed: NMS processing could slow down the inference. By removing it, the whole network could run on GPU without any CPU operations.
* Avoid hand-crafted hyper-parameters and design choices: the accuracy of the model could depend on how we select parameters (e.g. the number of boxes, sizes, aspect ratio). This could be biased and problem-specific.
* Context-aware prediction: by cropping rectangle boxes from the images as in the anchor-based approach, the prediction of the object dismissed the context of the object.

**Comment by glenn-jocher** [comment]
@biendltb thank you for your interest in YOLOv8 and for your insightful question about the advantages of anchor-free models. One of the biggest advantages of anchor-free models is that they are end-to-end differentiable, meaning that the network is trained end-to-end without any processing like NMS. This leads to better speed and allows the whole network to run purely on GPU without any CPU operations. Additionally, anchor-free models avoid hand-crafted hyperparameters and design choices which can be biased and problem-specific, allowing for a more analysis procedure in determining the best possible settings. Finally, with anchor-free models, more contextual information can be used in the prediction process, as the algorithm does not crop boxes away from the image as in the anchor-based approach.
